frontend/src/services/api/ApiNativeClient/index.ts

```diff
@@ -1,19 +1,42 @@
 import { ApiClient } from '../api';
-import { getCookie } from '../../../utils/cookies';
+import { systemClient } from '../../System';
 import NativeRobosats from '../../Native';
 
 class ApiNativeClient implements ApiClient {
-  constructor() {
-    if (!window.NativeRobosats) {
-      window.NativeRobosats = new NativeRobosats();
-    }
-  }
-
   private assetsCache: { [path: string]: string } = {};
   private assetsPromises: { [path: string]: Promise<string | undefined> } = {};
 
   private readonly getHeaders: () => HeadersInit = () => {
-    return { 'Content-Type': 'application/json', 'X-CSRFToken': getCookie('csrftoken') || '' };
+    let headers = {
+      'Content-Type': 'application/json',
+    };
+
+    const sessionid = systemClient.getCookie('sessionid');
+    if (sessionid) {
+      const robotToken = systemClient.getCookie('robot_token');
+      const csrftoken = systemClient.getCookie('csrftoken');
+      const pubKey = systemClient.getCookie('pub_key');
+
+      headers = {
+        ...headers,
+        ...{
+          'X-CSRFToken': csrftoken,
+          Cookie: `sessionid=${sessionid};robot_token=${robotToken};csrftoken=${csrftoken};pub_key=${pubKey}`,
```

Comment:
This was an unexpected issue waiting for me underneath 😄 POST requests on backend still require cookies to repsond